functionaltests/steps_v7.go

```diff
@@ -87,8 +87,6 @@ func emptyDataStreamsIngestV7(namespace string) esclient.DataStreamsDocCount {
 // NOTE: Only works for versions 7.x.
 type ingestV7Step struct{}
 
-var _ testStep = ingestV7Step{}
```

Comment:
Can you please merge `steps.go` and `steps_v7.go` and instead of having two files, organize things properly with `structs. This `steps_v7.go` has several things defined unrelated to `v7`, eg. [emptyIndicesIngest](https://github.com/elastic/apm-server/pull/16917/files#diff-b16cdc37470b93531b9fa9e7d9af76e3c6a757ff58c641b15ed9205c45a51785L33-R54), or the migration step, which can also happen on version 8.

Reply:
Even when support for `v7` stops, there could still be deployments that have not switched from standalone to managed. I'm not convinced we will be able to completely remove the test logic for this soon-ish. 
The logic can and should still be grouped together, however the way it is grouped now has some implicit knowledge in it that I am not sure is necessary. Let's discuss in our sync.